core/core/Array.hs

```diff
@@ -151,6 +158,13 @@ length :: Array a -> Int
 length = CL.lengthImpl
 
 
+-- | The indices that are valid for subscripting the collection, in ascending order.
+--
+-- > indices (fromLinkedList [1,2,3]) == [0,1,2]
+indices :: Array a -> [Int]
+indices = CL.indicesImpl
```

Comment:
Could you please flip the way you're doing this? I mean, implement the indices function here, and just do the linkage in `indicesImpl`? I know this is counterintuitive because of the Impl suffix, but it will make sense when I introduce traits formally and those will be just codegen'd

Reply:
You got it.  Thanks for the feedback!